pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -803,6 +821,14 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 		domain.Spec.MemoryBacking = &MemoryBacking{
 			HugePages: &HugePages{},
 		}
+		// virtiofs require shared access
+		if util.IsVMIVirtiofsEnabled(vmi) {
+			domain.Spec.MemoryBacking = &MemoryBacking{
+				Access: &MemoryBackingAccess{
+					Mode: "shared",
```

Comment:
I'm not sure I understand the hugepage requirement. This is using file-backed memory is it not? https://libvirt.org/kbase/virtiofs.html

from the doc, hugepage backed memory would look something more this?

```
<domain>
  ...
  <memoryBacking>
    <hugepages>
      <page size='2' unit='M'/>
    </hugepages>
    <access mode='shared'/>
  </memoryBacking>
  ...
</domain>
```

Reply:
This will mean that libvirt will need to allocate these hugepages inside the virt-launcher - which is not allowed from capabilities perspective.
Instead, our current approach with hugepages, in general, is to allocate the hugepages on the host and provide it to the guest as a filesystem mounted /dev/hugepages - in the future, we may switch to memfd [1] but this will require a larger change in kubevirt.

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1839030

Reply:
i see, thanks for the explanation.

Reply:
looking at this further, I'm not sure this is working as expected.

We do get hugepages within the pod, but it doesn't appear they are being used with the way this is currently configured.